sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -534,16 +566,20 @@ public void validate(PInput input) {
             verifyTablePresence(bqOptions, table);
           }
           if (query != null) {
-            dryRunQuery(bqOptions, query);
+            dryRunQuery(bqOptions, query, useLegacySql);
           }
         }
       }
 
-      private static void dryRunQuery(BigQueryOptions options, String query) {
+      private static void dryRunQuery(
+          BigQueryOptions options, String query, @Nullable Boolean useLegacySql) {
```

Comment:
No, it definitely can -- it's null if the user has not set it. Look ~10 lines up